File: Sources/FitUI/Components/FitPaymentMethodCard.swift
import SwiftUI

// MARK: - FitPaymentMethodCard
//
// Provider selector / connected-method card — leading 40×40 logo
// container + title + subtitle (with optional status dot) + trailing
// button or "Coming soon" badge. Used in payment / payout / balance
// settings. See `docs/components.md` § FitPaymentMethodCard.

public enum FitPaymentMethodStatus {
    case connected
    case actionRequired
    case notConnected
    case comingSoon
}

public struct FitPaymentMethodCard: View {
    let logoLetter: String
    let title: String
    let subtitle: String
    let status: FitPaymentMethodStatus
    let actionLabel: String?
    let onAction: (() -> Void)?
    let disabled: Bool
    let onTap: (() -> Void)?

    @Environment(\.fitTheme) private var theme

    public init(
        logoLetter: String,
        title: String,
        subtitle: String,
        status: FitPaymentMethodStatus = .notConnected,
        actionLabel: String? = nil,
        onAction: (() -> Void)? = nil,
        disabled: Bool = false,
        onTap: (() -> Void)? = nil
    ) {
        self.logoLetter = logoLetter
        self.title = title
        self.subtitle = subtitle
        self.status = status
        self.actionLabel = actionLabel
        self.onAction = onAction
        self.disabled = disabled
        self.onTap = onTap
    }

    public var body: some View {
        let row = HStack(spacing: FitSpacing.sp3) {
            // Logo container
            ZStack {
                RoundedRectangle(cornerRadius: FitRadius.sm)
                    .fill(theme.surfaceHigher)
                Text(String(logoLetter.prefix(1)).uppercased())
                    .font(.custom(FitFont.family, size: 18).weight(.semibold))
                    .foregroundColor(theme.textPrimary)
            }
            .frame(width: 40, height: 40)

            VStack(alignment: .leading, spacing: 2) {
                Text(title)
                    .font(.custom(FitFont.family, size: 15).weight(.medium))
                    .foregroundColor(theme.textPrimary)
                    .lineLimit(1)

                HStack(spacing: FitSpacing.sp1) {
                    if let dot = statusDotColor {
                        Circle()
                            .fill(dot)
                            .frame(width: 6, height: 6)
                    }
                    Text(subtitle)
                        .font(.custom(FitFont.family, size: 13))
                        .foregroundColor(theme.textTertiary)
                        .lineLimit(1)
                }
            }
            .frame(maxWidth: .infinity, alignment: .leading)

            trailing
        }
        .padding(.horizontal, 14)
        .padding(.vertical, FitSpacing.sp3)
        .background(theme.surfaceHigh)
        .clipShape(RoundedRectangle(cornerRadius: FitRadius.md))
        .opacity(disabled ? 0.6 : 1.0)

        if let onTap, !disabled {
            Button(action: onTap) { row }.buttonStyle(.plain)
        } else {
            row
        }
    }

    @ViewBuilder
    private var trailing: some View {
        switch status {
        case .comingSoon:
            FitBadge("Coming soon", style: .neutral)
        case .connected, .actionRequired, .notConnected:
            if let label = actionLabel, let onAction {
                FitButton(label, style: .secondary, size: .sm, action: onAction)
                    .fixedSize()
            }
        }
    }

    private var statusDotColor: Color? {
        switch status {
        case .connected:      return FitColors.Teal.t500
        case .actionRequired: return FitColors.Yellow.y400
        case .notConnected, .comingSoon: return nil
        }
    }
}

File: android/src/main/kotlin/com/fit321/fitui/components/FitPaymentMethodCard.kt
package com.fit321.fitui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.fit321.fitui.theme.LocalFitTheme
import com.fit321.fitui.tokens.FitColors
import com.fit321.fitui.tokens.FitFont
import com.fit321.fitui.tokens.FitRadius
import com.fit321.fitui.tokens.FitSpacing

// ============================================================================
// FitPaymentMethodCard — provider selector / connected method card.
// Mirrors Swift FitPaymentMethodCard. See `docs/components.md`
// § FitPaymentMethodCard.
// ============================================================================

enum class FitPaymentMethodStatus { Connected, ActionRequired, NotConnected, ComingSoon }

@Composable
fun FitPaymentMethodCard(
    logoLetter: String,
    title: String,
    subtitle: String,
    status: FitPaymentMethodStatus = FitPaymentMethodStatus.NotConnected,
    actionLabel: String? = null,
    onAction: (() -> Unit)? = null,
    disabled: Boolean = false,
    onTap: (() -> Unit)? = null,
    modifier: Modifier = Modifier
) {
    val theme = LocalFitTheme.current

    val statusDot: Color? = when (status) {
        FitPaymentMethodStatus.Connected -> FitColors.Teal.t500
        FitPaymentMethodStatus.ActionRequired -> FitColors.Yellow.y400
        else -> null
    }

    Row(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(FitRadius.md))
            .background(theme.surfaceHigh, RoundedCornerShape(FitRadius.md))
            .then(if (onTap != null && !disabled) Modifier.clickable(onClick = onTap) else Modifier)
            .padding(horizontal = 14.dp, vertical = FitSpacing.sp3)
            .alpha(if (disabled) 0.6f else 1f),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(FitSpacing.sp3)
    ) {
        // Logo container
        Box(
            modifier = Modifier
                .size(40.dp)
                .clip(RoundedCornerShape(FitRadius.sm))
                .background(theme.surfaceHigher),
            contentAlignment = Alignment.Center
        ) {
            Text(
                logoLetter.take(1).uppercase(),
                style = FitFont.heading3.copy(fontSize = 18.sp, fontWeight = FontWeight.SemiBold),
                color = theme.textPrimary
            )
        }

        Column(modifier = Modifier.weight(1f)) {
            Text(
                title,
                style = FitFont.body1.copy(fontSize = 15.sp, fontWeight = FontWeight.Medium),
                color = theme.textPrimary,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis
            )
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(FitSpacing.sp1)
            ) {
                if (statusDot != null) {
                    Box(modifier = Modifier.size(6.dp).clip(CircleShape).background(statusDot))
                }
                Text(
                    subtitle,
                    style = FitFont.footnote,
                    color = theme.textTertiary,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis
                )
            }
        }

        when (status) {
            FitPaymentMethodStatus.ComingSoon -> FitBadge("Coming soon", FitBadgeStyle.Neutral)
            else -> {
                if (actionLabel != null && onAction != null) {
                    // Inline secondary button (FitButton fills width — wrap so it
                    // sizes to content within the row).
                    Box(
                        modifier = Modifier
                            .clip(RoundedCornerShape(FitRadius.full))
                            .background(theme.surfaceHigher)
                            .clickable(onClick = onAction)
                            .padding(horizontal = FitSpacing.sp4, vertical = FitSpacing.sp2)
                    ) {
                        Text(
                            actionLabel,
                            style = FitFont.body2.copy(fontWeight = FontWeight.Medium),
                            color = theme.textPrimary
                        )
                    }
                }
            }
        }
    }
}
